src/cobalt.rs

```diff
@@ -115,44 +113,47 @@ pub fn build(config: &Config) -> Result<()> {
     // January 1, 1970 0:00:00 UTC, the beginning of time
     let default_date = UTC.timestamp(0, 0).with_timezone(&FixedOffset::east(0));
 
+    let (mut posts, documents): (Vec<Document>, Vec<Document>) = documents.into_iter()
+        .partition(|x| x.is_post);
+
     // sort documents by date, if there's no date (none was provided or it couldn't be read) then
     // fall back to the default date
-    documents.sort_by(|a, b| b.date.unwrap_or(default_date).cmp(&a.date.unwrap_or(default_date)));
+    posts.sort_by(|a, b| b.date.unwrap_or(default_date).cmp(&a.date.unwrap_or(default_date)));
 
     // check if we should create an RSS file and create it!
     if let &Some(ref path) = &config.rss {
-        try!(create_rss(path, dest, &config, &documents));
+        try!(create_rss(path, dest, &config, &posts));
     }
 
     // these are the attributes of all documents that are posts, so that they can be
     // passed to the renderer
-    // TODO: do we have to clone these?
-    let post_data: Vec<Value> = documents.iter()
-        .filter(|x| x.is_post)
+    let mut post_data: Vec<Value> = posts.iter()
         .map(|x| Value::Object(x.attributes.clone()))
         .collect();
 
-    // thread handles to join later
-    let mut handles = vec![];
+    let simple_post_data = post_data.clone();
 
-    // generate documents (in parallel)
-    crossbeam::scope(|scope| {
-        let post_data = Arc::new(post_data);
+    trace!("Generating posts");
+    for (post, data) in posts.iter().zip(post_data.iter_mut()) {
```

Comment:
**Awesome idea**, I think we can even improve on top of this and get down to cloning posts only once!